### PR TITLE
concept_is()

### PR DESCRIPTION
**Pull request by peterdutey**
Dear @AnikaC-git 

As discussed here's an additional wrapper function I would like to have reviewed.

```
> concepts_included_in(
+   concept_ids = "16227691000119107",  # Post-surgical excision site
+   target_ecl = "123037004"            # Body structure
+ )
[1] TRUE
> concepts_included_in(
+   concept_ids = "48800003",           # Ear lobule structure
+   target_ecl = "233604007"            # Pneumonia
+ )
[1] FALSE
> concepts_included_in(
+   concept_ids = "39732311000001104",  # Medical product only found UK Edition
+   target_ecl = "27658006"             # Product containing amoxicillin
+ )
[1] NA
```

Example use cases:
- recoding concepts to a small subset of higher-level (parent) concepts in a single step
- building classifications of concepts based on mutually-exclusive categories.
- determining whether a concept 'is' something that doesn't currently exist in SNOMED CT, yet can be defined by an ECL statement

The function does two things
1. checks whether the input exists as a concept on the snowstorm branch (it may not exist if mistyped, or if only exists on a different branch/national edition)
2. checks whether the input is a descendant of a target concept or ECL statement

Hope it all makes sense. Note that this is not something that exists in Snowstorm as a single step.

**Question** I can't seem to find a good name for this function.
Originally I named it `concept_is()` but not very self-explanatory.
Now `concept_included_in()` but it might be worse still?
Any suggestion welcome.
FYI other wrapper functions are named
* concepts_find
* concepts_ancestors
* concepts_descendants
* concepts_descriptions
* concepts_map

Many thanks in advance for the review.
Peter

### CODE REVIEW

File: tests/testthat/test.wrapper.R [hunk 1]
```diff
@@ -202,6 +202,90 @@ test_that("concepts_map no equivalent", {
 })
 
 
+
+# concepts_included_in ----------------------------------------------------
+
+test_that("concepts_included_in", {
```
Comment (AnikaC-git) [inline tests/testthat/test.wrapper.R:208]: if one were to aim for a complete set of tests, I guess it might be helpful to test the validation of inputs too (as in empty ecl) and also include a test for the progress bar progress; I might have not seen the empty ecl though ...

**Review by AnikaC-git — APPROVED** [review]
Hi @peterdutey ,

This additional function looks very useful to me and hope that users will appreciate it too. As for naming, I prefer to be on the edge of just about acceptable length and might even expand the name to be "concepts_in_ecl_query_results" or similar to express the functionality behind it. I personally would shy away from "..._in_ancestors" and "..._in_descendants" as someone having worked with ontologies may expect very specific behaviour and in consequence be less appreciative if the function doesn't behave in the expected way. 

While the code looks perfectly fine, the only comments I made was in relation to perhaps modularise the code a bit more and expand the test cases a little, but neither is exactly crucial to operation.

Any questions in relation to my comments, please let me know.

File: R/wrapper.R [hunk 2]
```diff
@@ -122,6 +122,143 @@ concepts_find <- function(term = NULL,
 }
 
 
+#' Determine where a concept of interest belongs to a target set of concepts
```
[the remark this answers is not in the document]
Comment (peterdutey) [reply R/wrapper.R]: https://github.com/ramses-antibiotics/snomedizer/pull/21/commits/667154f216e1462a04413fdd6e87bb6a37260866

File: R/wrapper.R [hunk 3]
```diff
@@ -122,6 +122,143 @@ concepts_find <- function(term = NULL,
...
+  branch = snomedizer_options_get("branch"),
+  encoding = "UTF-8"
+) {
+
+  stopifnot(length(target_ecl) == 1)
+
+  unique_concept_ids <- as.character(stats::na.omit(concept_ids))
+  # remove empty strings
+  unique_concept_ids <- grep(pattern = "^$",
+                             x = unique_concept_ids,
```
[the remark this answers is not in the document]
Comment (peterdutey) [reply R/wrapper.R]: https://github.com/ramses-antibiotics/snomedizer/pull/21/commits/6968fa253848b97c2a9b118f66d96b1e6c53fa44

File: R/wrapper.R [hunk 4]
```diff
@@ -122,6 +122,143 @@ concepts_find <- function(term = NULL,
...
+
+  if( is.null(valid_concepts) ) {
+    return(rep(as.logical(NA), length(concept_ids)))
+  }
+
+  if( !silent ) {
+    progress_bar <- progress::progress_bar$new(
+      format = "  [:bar] :percent :eta",
+      total = (trunc(length(unique_concept_ids)/200) + 1)
+    )
```
[the remark this answers is not in the document]
Comment (peterdutey) [reply R/wrapper.R]: https://github.com/ramses-antibiotics/snomedizer/pull/21/commits/daf8553d6125e07308542a088bbd4e6640bb4296

File: tests/testthat/test.wrapper.R (hunk 1 above)
Comment (peterdutey) [reply tests/testthat/test.wrapper.R:208]: https://github.com/ramses-antibiotics/snomedizer/pull/21/commits/daf8553d6125e07308542a088bbd4e6640bb4296

File: R/wrapper.R [hunk 5]
```diff
@@ -122,6 +122,143 @@ concepts_find <- function(term = NULL,
...
+    progress_bar$tick(0)
+  } else {
+    progress_bar <- NULL
+  }
+
+  # split into batches of 200 concepts
+  unique_concept_ids <- split(
+    unique_concept_ids,
+    sort(trunc(seq_len(length(unique_concept_ids))/200))
+  )
```
[the remark this answers is not in the document]
Comment (peterdutey) [reply R/wrapper.R]: https://github.com/ramses-antibiotics/snomedizer/pull/21/commits/daf8553d6125e07308542a088bbd4e6640bb4296

File: R/wrapper.R [hunk 6]
```diff
@@ -122,6 +122,143 @@ concepts_find <- function(term = NULL,
...
+#' @param target_ecl character ECL expression defining the target set of concepts
+#' @param silent whether to hide progress bar. Default is \code{FALSE}
+#' @param endpoint URL of a SNOMED CT Terminology Server REST API endpoint.
+#'  See \code{\link{snomedizer_options}}.
+#' @param branch a string for the name of the API endpoint branch to use (most
+#' commonly \code{"MAIN"}). See \code{\link{snomedizer_options}}.
+#' @param encoding HTTP charset parameter to use (default is \code{"UTF-8"})
+#' @return a logical vector with the same order as \code{concept_ids} indicating
+#' whether each concept is included in the target set or not. \code{NA} denotes
+#' a REST error.
```
[the remark this answers is not in the document]
Comment (peterdutey) [reply R/wrapper.R]: https://github.com/ramses-antibiotics/snomedizer/pull/21/commits/667154f216e1462a04413fdd6e87bb6a37260866
